### DISCUSSION TITLE
How to set Resource limits for metaMonitoring?

### DISCUSSION THREAD

**Discussion by jonli662** [grafana/mimir#6973] [Help and support]
Hi,
Trying to deploy Mimir (distributed) with the [Helm chart](https://github.com/grafana/mimir/tree/mimir-distributed-5.1.4/operations/helm/charts/mimir-distributed), but I'm getting the following error:

```
k get events
LAST SEEN   TYPE      REASON         OBJECT                               MESSAGE
11m         Warning   FailedCreate   statefulset/mimir-meta-monitoring    create Pod mimir-meta-monitoring-0 in StatefulSet mimir-meta-monitoring failed error: pods "mimir-meta-monitoring-0" is forbidden: failed quota: xxx-resource-quota: must specify limits.cpu,limits.memoryWe have a policy within our clusters that demands specifying limits on cpu and memory.
```

But HOW do I supply limits-values for the "metaMonitoring:-block" in the Helm chart (all other "blocks" have resources-variables for this)?

**Comment by dimitarvdimitrov** [comment]
I think this needs to be done via a change in the GrafanaAgent custom resource that the mimir-distributed helm chart creates. The CRD exposes the ability to [change the resources of the containers](https://doc.crds.dev/github.com/grafana/helm-charts/monitoring.grafana.com/GrafanaAgent/v1alpha1@grafana-agent-operator-0.2.15#spec-containers-resources). The mimir-distributed helm chart already does this for security contexts for the agent -  [example](https://github.com/grafana/mimir/blob/9c8cc8a5df8934d929e6b587db6c4065a00a6327/operations/helm/charts/mimir-distributed/templates/metamonitoring/grafana-agent.yaml#L23-L44). 


Contributions are welcome :)
